bindings/alicloud/nacos/nacos.go

```diff
@@ -0,0 +1,426 @@
+	watches      []configParam
+	servers      []constant.ServerConfig
+	logger       logger.Logger
+	configClient config_client.IConfigClient
+	readHandler  func(response *bindings.ReadResponse) ([]byte, error)
+}
+
+// NewNacos returns a new Nacos instance
+func NewNacos(logger logger.Logger) *Nacos {
+	return &Nacos{logger: logger} //nolint:exhaustivestruct
+}
+
+// Init implements InputBinding/OutputBinding's Init method
+func (n *Nacos) Init(metadata bindings.Metadata) (err error) {
+	if n.init {
+		return nil
+	}
+	n.init = true
+	n.name = metadata.Name
+
+	n.metadata, err = parseMetadata(metadata)
+	if err != nil {
+		err = fmt.Errorf("nacos config error: %w", err)
+
+		return
+	}
+
+	if n.metadata.TimeoutMs <= 0 {
+		n.metadata.TimeoutMs = 10000
+	}
+
+	if n.metadata.Endpoint != "" {
+		n.logger.Infof("nacos server's url: %s", n.metadata.Endpoint)
+	} else if n.metadata.NameServer != "" {
+		n.logger.Infof("nacos nameserver: %s", n.metadata.NameServer)
+	} else {
+		err = errors.New("nacos config error: must config endpoint or nameserver")
+
+		return
+	}
+
+	if n.metadata.Config != "" {
+		n.config, err = convertConfig(n.metadata.Config)
+		if err != nil {
+			return
+		}
+	}
+	n.watches, err = convertConfigs(n.metadata.Watches)
+	if err != nil {
+		return
+	}
+
+	n.servers, err = convertServers(n.metadata.Endpoint)
+	if err != nil {
+		return
+	}
+
+	return n.createConfigClient()
+}
+
+func (n *Nacos) createConfigClient() error {
+	config := map[string]interface{}{}
+	config["clientConfig"] = constant.ClientConfig{ //nolint:exhaustivestruct
+		TimeoutMs:            uint64(n.metadata.TimeoutMs),
+		NamespaceId:          n.metadata.NamespaceID,
+		Endpoint:             n.metadata.NameServer,
+		RegionId:             n.metadata.RegionID,
+		AccessKey:            n.metadata.AccessKey,
+		SecretKey:            n.metadata.SecretKey,
+		OpenKMS:              n.metadata.AccessKey != "" && n.metadata.SecretKey != "",
+		CacheDir:             n.metadata.CacheDir,
+		UpdateThreadNum:      n.metadata.UpdateThreadNum,
+		NotLoadCacheAtStart:  n.metadata.NotLoadCacheAtStart,
+		UpdateCacheWhenEmpty: n.metadata.UpdateCacheWhenEmpty,
+		Username:             n.metadata.Username,
+		Password:             n.metadata.Password,
+		LogDir:               n.metadata.LogDir,
+		RotateTime:           n.metadata.RotateTime,
+		MaxAge:               int64(n.metadata.MaxAge),
+		LogLevel:             n.metadata.LogLevel,
+	}
+
+	if len(n.servers) > 0 {
+		config["serverConfigs"] = n.servers
+	}
+
+	var err error
+	n.configClient, err = clients.CreateConfigClient(config)
+	if err != nil {
+		return fmt.Errorf("nacos config error: create config client failed. %w ", err)
+	}
+
+	return nil
+}
+
+// Read implements InputBinding's Read method
+func (n *Nacos) Read(handler func(*bindings.ReadResponse) ([]byte, error)) error {
+	n.readHandler = handler
+
+	for _, watch := range n.watches {
+		go n.startListen(watch)
+	}
+
+	exitChan := make(chan os.Signal, 1)
```

Comment:
I've seen this in a few components now and its less than ideal but the only way bindings can cleanup at the moment . Just pointing out this issue related to using a `Close() error` method for bindings. #779. I may implement this in `dapr/dapr` so you can use a context for cancelation.

Reply:
dapr/dapr#3088 make it so `Close() error` will be called on components. Please use a component level context for cancelation (see the Kafka pubsub for an example)

Reply:
@cinience dapr/dapr#3088 was merged. Please implement `io.Closer` in this component.